### PR TITLE
power of three problem solved

### PR DESCRIPTION
**Pull request by unicode368**

### CODE REVIEW

File: src/PowerOfThree.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+public class PowerOfThree {
+    public static boolean isPowerOfThree(int n) {
+        if(n <= 0) {
+            return false;
+        }
+
+        while(n % 3 == 0 && n > 3) {
```
Comment (ybystrov) [inline src/PowerOfThree.java]: You can change condition to while(n % 3 == 0 && n >= 3), then return can be:
return n == 1

File: test/PowerOfThreeTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        return Arrays.asList(new Object[][] {
+                { 27 , true, "Three to the power of 3" },
+                { 19647 , false, "Aliquot but not power of" },
+                { 1 , true, "Power of 3 to 0" },
+                { -3 , false, "Not the power of three" }
+        });
+    }
+
+    @Test
+    public void testAddTwoNumbers() {
```
Comment (ybystrov) [inline test/PowerOfThreeTest.java]: Why is the test called "testAddTwoNumbers"?

File: test/PowerOfThreeTest.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    public PowerOfThreeTest(int number, boolean expectedResult,
+                        String description) {
+        this.number = number;
+        this.expectedResult = expectedResult;
+        this.description = description;
+    }
+
+    @Parameterized.Parameters
+    public static Collection numsAndTargets() {
```
Comment (ybystrov) [inline test/PowerOfThreeTest.java]: Use JUnit5, which has better way of writing parameterised tests. 
Have a look here: 
https://stackoverflow.com/questions/61483452/parameterized-test-with-two-arguments-in-junit-5-jupiter

File: test/PowerOfThreeTest.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+                { 27 , true, "Three to the power of 3" },
+                { 19647 , false, "Aliquot but not power of" },
+                { 1 , true, "Power of 3 to 0" },
+                { -3 , false, "Not the power of three" }
+        });
+    }
+
+    @Test
+    public void testAddTwoNumbers() {
+        System.out.println(description);
```
Comment (ybystrov) [inline test/PowerOfThreeTest.java]: I assume you write description to console to know which test is it. 
Using JUnit5, you can do it more easily by formatting name of the test.
Example:

@ParameterizedTest(name = "{0} + {1} == {2}")
    @MethodSource("getAddparams")
    void testAdd(int x, int y, int expRes) {
        var calc = new Calc();
        assertEquals(expRes, calc.add(x, y));
    }

File: test/PowerOfThreeTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        this.description = description;
+    }
+
+    @Parameterized.Parameters
+    public static Collection numsAndTargets() {
+        return Arrays.asList(new Object[][] {
+                { 27 , true, "Three to the power of 3" },
+                { 19647 , false, "Aliquot but not power of" },
+                { 1 , true, "Power of 3 to 0" },
+                { -3 , false, "Not the power of three" }
```
Comment (ybystrov) [inline test/PowerOfThreeTest.java]: I would also add number 0 as an edge case

File: src/PowerOfThree.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+public class PowerOfThree {
+    public static boolean isPowerOfThree(int n) {
```
Comment (ybystrov) [inline src/PowerOfThree.java:2]: Very good. I like this implementation

File: test/PowerOfThreeTest.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class PowerOfThreeTest {
+
+    private static Stream<Arguments> provideParameters() {
```
Comment (ybystrov) [inline test/PowerOfThreeTest.java:17]: Very good. Like it!

**Review by ybystrov — APPROVED** [review]
